cli/aws-cron-validator.ts

```diff
@@ -0,0 +1,109 @@
+class AWSCronError extends Error {}
+
+export class AWSCronValidator {
+
+    private static minuteExp = `(0?[0-9]|[1-5][0-9])`; // [0]0-59
+    private static hourExp = `(0?[0-9]|1[0-9]|2[0-3])`; // [0]0-23
+    private static dayOfMonthExp = `(0?[1-9]|[1-2][0-9]|3[0-1])`; // [0]1-31
+    private static monthExp = `(0?[1-9]|1[0-2]|JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)`; // [0]1-12 or JAN-DEC
+    private static dayOfWeekExp = `([1-7]|SUN|MON|TUE|WED|THU|FRI|SAT)`; // 1-7 or SAT-SUN
+    private static dayOfWeekHash(): string {
+        return `(${this.dayOfWeekExp}#[1-5])`; // add hash expression to enable supported use case
+    }
+    private static yearExp = `((19[8-9][0-9])|(2[0-1][0-9][0-9]))`; // 1980-2199
+    private static numbers = `([0-9]*[1-9][0-9]*)`; // whole numbers greater than 0
+
+    public static validate(expression: string): string {
+        if (!expression.trim()) {
+            throw new AWSCronError(
+                `No parameters entered, this format is required in UTC: 0 20 ? * SUN-FRI *`
+            );
+        }
+        const valueCount = expression.split(" ").length;
+        if (valueCount !== 6) {
+          throw new AWSCronError(
+              `Incorrect amount of parameters in '${expression}'. 6 required, ${valueCount} provided.`
+          );
+        }
+
+        const [minute, hour, dayOfMonth, month, dayOfWeek, year] = expression.split(" ");
+
+        // special handling for Day of Month and Day of Week
+        if (!((dayOfMonth === "?" && dayOfWeek !== "?") || (dayOfMonth !== "?" && dayOfWeek === "?"))) {
+            throw new AWSCronError(
+                `Invalid combination of day-of-month '${dayOfMonth}' and day-of-week '${dayOfWeek}'. One must be a question mark (?)`
+            );
+        }
+
+        if (!new RegExp(this.minuteRegex()).test(minute)) {
+            throw new AWSCronError(`Invalid minute value '${minute}'.`);
+        }
+        if (!new RegExp(this.hourRegex()).test(hour)) {
+            throw new AWSCronError(`Invalid hour value '${hour}'.`);
+        }
+        if (!new RegExp(this.dayOfMonthRegex()).test(dayOfMonth)) {
+            throw new AWSCronError(`Invalid day-of-month value '${dayOfMonth}'.`);
+        }
+        if (!new RegExp(this.monthRegex(), 'i').test(month)) {
+            throw new AWSCronError(`Invalid month value '${month}'.`);
+        }
+        if (!new RegExp(this.dayOfWeekRegex(), 'i').test(dayOfWeek)) {
+            throw new AWSCronError(`Invalid day-of-week value '${dayOfWeek}'.`);
+        }
+        if (!new RegExp(this.yearRegex()).test(year)) {
+            throw new AWSCronError(`Invalid year value '${year}'.`);
+        }
+
+        return expression;
+    }
+
+    // Build up regex's incrementally improving readability of the above expressions
+    private static rangeRegex(values: string): string {
```

Comment:
nit: I would consider getting these to return `RegExp` objects to save you wrapping a lot of the constructor calls above when you get strings out here. For the cases where you need to re-use the string value as part of another regex below you can reference that like `this.rangeRegex.source` which is still pretty clean